scripts/tests/core-integration.test.mjs

```diff
@@ -195,253 +190,81 @@ describe("CORE Integration Tests", () => {
+        const isReadme = basename === "README.md";
+        const heading = sourceContent.match(/#[\s]+(.*)/);
+        const title = isReadme ? "Overview" : heading[1];
+        const sidebarLabel = isReadme ? "\nsidebar:\n  label: Overview" : "";
 
-      await fs.mkdir(path.dirname(sourceFile), { recursive: true });
-      await fs.writeFile(sourceFile, sourceContent, "utf8");
+        expect(title).toBe(expectedTitle);
+        if (shouldHaveSidebar) {
+          expect(sidebarLabel).toContain("sidebar:");
+          expect(sidebarLabel).toContain("label: Overview");
+        } else {
+          expect(sidebarLabel).toBe("");
+        }
+      },
+    );
 
-      // Simulate single file mapping
+    it.each([
+      ["best-practices/README.md", "best-practices", "reference/best-practices.md"],
+      ["module-examples/README.md", "module-examples", "reference/module-examples.md"],
+      ["faq/README.md", "faq", "reference/faq.md"],
+    ])("should map special directory %s to single file", async (inputPath, dir, expectedOutput) => {
       const PATH_MAPPINGS = {
         singleFile: {
           "best-practices": "reference/best-practices.md",
+          "module-examples": "reference/module-examples.md",
+          faq: "reference/faq.md",
         },
       };
 
-      const relativePath = path.relative(docsDir, sourceFile);
-      const dir = path.dirname(relativePath);
-      const basename = path.basename(relativePath);
-
-      let outputPath = relativePath;
+      const basename = path.basename(inputPath);
+      let outputPath = inputPath;
       if (basename === "README.md" && PATH_MAPPINGS.singleFile[dir]) {
         outputPath = PATH_MAPPINGS.singleFile[dir];
       }
 
-      expect(outputPath).toBe("reference/best-practices.md");
-    });
-  });
-
-  describe("Error Handling", () => {
-    it("should handle missing files gracefully", async () => {
-      const missingFile = path.join(docsDir, "nonexistent.md");
-
-      let hasError = false;
-      try {
-        await fs.access(missingFile);
-      } catch {
-        hasError = true;
-      }
-
-      expect(hasError).toBe(true);
-    });
-
-    it("should handle empty content files", async () => {
-      const emptyFile = path.join(docsDir, "empty.md");
-      await fs.writeFile(emptyFile, "", "utf8");
-
-      const content = await fs.readFile(emptyFile, "utf8");
-      expect(content).toBe("");
-
-      const heading = content.match(/#[\s]+(.*)/);
-      expect(heading).toBeNull();
-    });
-
-    it("should handle malformed markdown", async () => {
-      const malformedFile = path.join(docsDir, "malformed.md");
-      const content = "This has no heading\n\nJust content...";
-      await fs.writeFile(malformedFile, content, "utf8");
-
-      const fileContent = await fs.readFile(malformedFile, "utf8");
-      const heading = fileContent.match(/#[\s]+(.*)/);
-
-      expect(heading).toBeNull();
+      expect(outputPath).toBe(expectedOutput);
```

Comment:
This also seems like a test case where we define the inputs and outputs prior to the test and assert that the outputs haven't changed without acting upon them. In this case, I'd ask myself how I might make a code change in the "production" code that could cause this test to fail (which I think means somewhere in `scripts/lib/`?). If there's not a way to do that, this test case may provide false-confidence.